README.md

```diff
@@ -1,207 +1,304 @@
-
-## Get Started
-```python
-import GreenplumPython as gp
-import Pandas as pd
-
-# init connection
-connection = gp.Connection()
-
-# connect to GPDB
-conn = connection.connect(dbIP, dbPort, dbname, username, password, ...)
+GreenplumPython is a Python library that enables the user to interact with Greenplum in a Pythonic way.
 
-# init GPDB Database instance
-GPDatabase_instance = gp.GPDatabase(conn)
+GreenplumPython provides a pandas-like table API that
+1. looks familiar and intuitive to Python users
+2. is powerful to do complex analytics, such as statistical analysis, with UDFs and UDAs
+3. encapsulates common best practices and avoids common pitfalls in Greeenplum, compared to writing SQL directly
 
-# a dataframe warpper for input table
-Dataframe_Wrapper_instance_input = gpdb_instance.get_table(schema, table_name)
+# Selecting the Database of Your Data
 
-# set output table columns info 
-columns_output_types = list()
-column_type_tuple = ('a', 'int4')
-columns_output_types.append(column_type_tuple)
+To begin with, we need to select the database that contains the data we want:
 
-# set output table info without need to store results in a new table
-Table_Metadata_output = gp.GPTableMeta(None, None, columns_output_types, None)
 
-# apply function
-def input_py_func(row_element, extra_args):
-    return row_element + extra_args
-
-# run gpapply and return result is a pandas dataframe
-dataframe = gp.gpapply(Dataframe_Wrapper_instance_input, input_py_func, GPDatabase_instance, Table_Metadata_output, clear_existing = True, runtime_id = 'plc_r', runtime_type = 'plcontainer', **input_py_func_extra_args)
-
-# group apply function
-def input_py_func_groupby(pd.dataframe, extra_args):
-    import Pandas as pd
-    for row in pd.dataframe
-        yield row['a'] + extra_args
-
-# set output table info and store results in a new table
-Table_Metadata_output = gp.GPTableMeta("new_table", "schema", columns_output_types, None)
+```python
+import greenplumpython as gp
 
-# run gpTapply
-gp.gpTapply(Dataframe_Wrapper_instance_input, input_py_func_groupby, group_by_index, GPDatabase_instance, Table_Metadata_output = None, clear_existing = True, runtime_id = 'plc_r', runtime_type = 'plcontainer', **input_py_func_extra_args)
 
+db = gp.database(host="localhost", dbname="gpadmin")
 ```
-## Main Funcions Usage
-
-### gpApply: 
- Apply a R function to every row of data of a data frame object.
 
+We will use the following utility function to display a table:
+```python
+from tabulate import tabulate
 
-### Description
 
- gpapply allows you to run a R function with input data frame in GreenplumDB through PL/container or PL/Python language.
- The Python function will be parsed, then an UDF will be created in GreenplumDB schema for execution.
- The calculation will be done in parallel with computing resources of GPDB segment hosts.
+def display(t: gp.Table):
+    return tabulate(t.fetch(), headers="keys", tablefmt="html")
```

Comment:
It might be good to mention we are using the `jupyter notebook` or `jupyter lab` environment?

Reply:
Yes, I mean the `display()` function cannot print the table under other environment, e.g., in a terminal? It's ok to leave it as is :-)